From 9cf3c9dcefcd8c136725e2a40128166cd52d20ca Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Wed, 22 Jul 2026 14:28:49 +0100
Subject: [PATCH] tests/relabel/test-setfiles.py: Add a test of setfiles
 excludes list

Add a test that we don't relabel directories listed in the excludes
list.

(cherry picked from commit 44d271a80308bb3898f48662aace25937481688b)
---
 tests/relabel/test-setfiles.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tests/relabel/test-setfiles.py b/tests/relabel/test-setfiles.py
index 6b080d42a..0b0eb4966 100755
--- a/tests/relabel/test-setfiles.py
+++ b/tests/relabel/test-setfiles.py
@@ -62,6 +62,8 @@ g.mount_options("user_xattr", "/dev/sda1", "/")
 g.mkdir("/bin")
 g.touch("/bin/ls")
 g.mkdir("/etc")
+g.mkdir("/lib")
+g.touch("/lib/libc.so")
 g.mkdir("/tmp")
 g.touch("/tmp/test")
 g.mkdir("/var")
@@ -75,13 +77,14 @@ g.write("/etc/file_contexts", """/.* system_u:object_r:default_t:s0
 /bin/.* system_u:object_r:bin_t:s0
 /etc/.* system_u:object_r:etc_t:s0
 /etc/file_contexts <<none>>
+/lib/.* system_u:object_r:lib_t:s0
 /tmp/.* <<none>>
 /var/.* system_u:object_r:var_t:s0
 /var/log/.* system_u:object_r:var_log_t:s0
 """)
 
 # Do the relabel.
-g.setfiles("/etc/file_contexts", "/", force=True)
+g.setfiles("/etc/file_contexts", "/", force=True, excludes=["/lib", "/lib64"])
 
 # Check the labels were set correctly.
 errors = 0
@@ -122,6 +125,8 @@ check_label("/bin", "system_u:object_r:default_t:s0")
 check_label("/bin/ls", "system_u:object_r:bin_t:s0")
 check_label("/etc", "system_u:object_r:default_t:s0")
 check_label_none("/etc/file_contexts")
+check_label_none("/lib") # because /lib in excludes
+check_label_none("/lib/libc.so") # because /lib in excludes
 check_label("/tmp", "system_u:object_r:default_t:s0")
 check_label_none("/tmp/test")
 check_label("/var", "system_u:object_r:default_t:s0")
-- 
2.47.3

